test/fakeintake/server/testhelper.go

```diff
@@ -21,7 +21,8 @@ func InitialiseForTests(t *testing.T, opts ...Option) (*Server, *clock.Mock) {
 	t.Helper()
 	ready := make(chan bool, 1)
 	mockClock := clock.NewMock()
-	opts = append(opts, WithReadyChannel(ready), WithClock(mockClock), WithAddress("127.0.0.1:0"))
+	// Add the options at the end so they can override the defaults
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**